### DISCUSSION TITLE
Use as an external data repository? (R-specific)

### DISCUSSION THREAD

**Discussion by mhpob** [ocean-tracking-network/latitudes#14] [Ideas]
This article outlines in great detail how to remove internal data to an external repository in order to get a package under the 5MB CRAN limit.

https://journal.r-project.org/archive/2017/RJ-2017-026/index.html

I imagine this repo may seek to be language agnostic; perhaps that, in addition to testing, an R-specific branch of this repo that is tagged for R-universe to build binaries could fill the niche outlined in that article (rather than drat)?

Directly applicable to the `glatos`-to-CRAN journey, paging @chrisholbrook.

**Comment by jdpye** [comment]
My thought is that this primarily will kick tires on https://github.com/ocean-tracking-network/surimi but yeah, will serve as a repo for general industry-standard file types and ensuring that we or anyone who cares to can handle them properly.

**Comment by mhpob** [comment]
Two birds, one stone! An example if this gets pondered in the future:

Repositories using R-Universe in the Additional_repositories field:
https://github.com/search?q=Additional_repositories+r-universe&type=code&utf8=%E2%9C%93

One example on CRAN is qgisprocess: https://github.com/r-spatial/qgisprocess
  DESCRIPTION: https://github.com/r-spatial/qgisprocess/blob/main/DESCRIPTION#L62
  CRAN comments: https://github.com/r-spatial/qgisprocess/blob/main/cran-comments.md?plain=1#L25
  The package it references: https://geocompr.r-universe.dev/spDataLarge; https://github.com/Nowosad/spDataLarge

**Comment by chrisholbrook** [comment]
> serve as a repo for general industry-standard file types and ensuring that we or anyone who cares to can handle them properly.

Based on #2, it seems the primary purpose of this repo is a repo of industry-standard files, which I fully support. If the "second bird" is a repo of example data for demonstrating functions of various packages, then I have some reservations, simply because we've done some scoping on both goals for GLATOS and they seem to require separate criteria/characteristics. E.g., the example data currently packages with glatos (for running example code in documentation) was selected mostly by convenience; we intend to go back and identify characteristic datasets with a useful level of complexity and that can be used across several functions, but haven't attempted it yet.

**Comment by mhpob** [comment]
I'm not sure they're mutually exclusive goals. Using `glatos` as an example, `inst/extdata` is around 34 MB. `/detection_files_raw` and direct derivatives are 4.5 MB. The CRAN size limit for a package is 5 MB. If the desire is to get into CRAN, these files need to be offloaded somehow. So... while there may be, admittedly, 30 more MB to go here, it still is offloading a suite of industry-standard files that amount to the maximum size of a package.

There will obviously be package-specific needs that would be beyond this scope, but there would be coverage for a large amount of testing and use cases.

**Comment by benjaminhlina** [comment]
I've started adding some of my own data to this in a separate [fork](https://github.com/benjaminhlina/demo-data/tree/add-range-data) on my forked version of the repo. I, however, may have gotten lost on the idea of what types of data we are wanting to include...I've added under raw a .vrl file and have started updating the metadata information (not complete at this moment). Is this something we are thinking of including? I was also thinking to include a csv version of this file in separate directory within the repo as I'd like to reference the file and use it in the detection range vignette to demonstrate how to calculate preliminary detection efficiency. If the csv version of this doesn't fit the scope of the repo that is understandable. Appreciate all the heavy lifting that has occurred to get this going....let me know if I'm heading in the right direction with all of this...thanks :)
